spec/cypress/integration/admin/suppliers/create-lead-provider-user.js

```diff
@@ -0,0 +1,142 @@
+    cy.chooseLeadProviderName();
+
+    cy.location("pathname").should(
+      "equal",
+      "/admin/suppliers/users/new/user-details"
+    );
+    cy.chooseNameAndEmail(userName, userEmail);
+
+    cy.location("pathname").should(
+      "equal",
+      "/admin/suppliers/users/new/review"
+    );
+    cy.get("main").should("contain", userName);
+    cy.get("main").should("contain", userEmail);
+    cy.confirmCreateSupplierUser();
+
+    cy.location("pathname").should(
+      "contain",
+      "/admin/suppliers/users/new/success"
+    );
+
+    cy.get("a").contains("manage suppliers").click();
+    cy.location("pathname").should("equal", "/admin/suppliers");
+
+    cy.get("a").contains("All users").click();
+    cy.location("pathname").should("equal", "/admin/suppliers/users");
+    cy.get("main").should("contain", userName);
+    cy.get("main").should("contain", userEmail);
+    cy.appEval(`LeadProvider.first.name`).then((leadProviderName) =>
+      cy.get("main").should("contain", leadProviderName)
+    );
+
+    cy.appEval(`User.find_by(email: "${userEmail}").present?`).then((result) =>
+      expect(result).to.equal(true)
+    );
+    cy.appEval(
+      `User.find_by(email: "${userEmail}").lead_provider.present?`
+    ).then((result) => expect(result).to.equal(true));
+  });
+
+  it("remembers previous choices", () => {
+    cy.appScenario("admin/suppliers/create_supplier");
+    const userName = "John Smith";
+    const userEmail = "j.s@example.com";
+
+    cy.visit("/admin/suppliers/users");
+    cy.get(".govuk-button").click();
+
+    cy.chooseLeadProviderName();
+    cy.clickBackLink();
+    cy.appEval(`LeadProvider.first.name`).then((leadProviderName) => {
+      cy.get("input[type=text]").should("have.value", leadProviderName);
+      cy.get(".govuk-button").click();
+    });
+
+    cy.chooseNameAndEmail(userName, userEmail);
+    cy.clickBackLink();
+    cy.get("input[name='supplier_user_form[full_name]'").should(
+      "have.value",
+      userName
+    );
+    cy.get("input[name='supplier_user_form[email]'").should(
+      "have.value",
+      userEmail
+    );
+    cy.get(".govuk-button").click();
+
+    cy.confirmCreateSupplierUser();
+
+    cy.get("a").contains("manage suppliers").click();
+    cy.location("pathname").should("equal", "/admin/suppliers");
+
+    cy.get("a").contains("All users").click();
+    cy.location("pathname").should("equal", "/admin/suppliers/users");
+    cy.get("main").should("contain", userName);
+    cy.get("main").should("contain", userEmail);
+    cy.appEval(`LeadProvider.first.name`).then((leadProviderName) =>
+      cy.get("main").should("contain", leadProviderName)
+    );
+
+    cy.appEval(`User.find_by(email: "${userEmail}").present?`).then((result) =>
+      expect(result).to.equal(true)
+    );
+    cy.appEval(
+      `User.find_by(email: "${userEmail}").lead_provider.present?`
+    ).then((result) => expect(result).to.equal(true));
+  });
+
+  it.only("allows changing name choice", () => {
```

Comment:
this looks like it was left in by mistake

(we should probably add https://www.npmjs.com/package/eslint-plugin-no-only-tests at some point - the single greatest linting rule there is)